tests/Api/Category/ListCategoryIntegration.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Akeneo\Pim\tests\Api\Category;
+
+use Akeneo\Pim\Api\CategoryApi;
+use Akeneo\Pim\Pagination\PageInterface;
+use Akeneo\Pim\Pagination\ResourceCursorInterface;
+use Akeneo\Pim\tests\Api\ApiTestCase;
+
+class ListCategoryIntegration extends ApiTestCase
+{
+    public function testListPerPage()
+    {
+        $api = $this->createClient()->getCategoryApi();
+        $uriGenerator = $this->getUriGenerator();
+        $expectedCategories = $this->getExpectedCategories();
+
+        $firstPage = $api->listPerPage(2);
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertNull($firstPage->getCount());
+        $this->assertNull($firstPage->getPreviousLink());
+        $this->assertNull($firstPage->getPreviousPage());
+
+        $this->assertSame($uriGenerator->generate(CategoryApi::CATEGORIES_PATH, [], [
+            'page' => 2,
+            'limit' => 2,
+            'with_count' => false,
+        ]), $firstPage->getNextLink());
+
+        $categories = $firstPage->getItems();
+        $this->assertCount(2 ,$categories);
+        $this->assertSameContent($expectedCategories[0], $categories[0]);
+        $this->assertSameContent($expectedCategories[1], $categories[1]);
+
+        $secondPage = $firstPage->getNextPage();
+        $this->assertInstanceOf(PageInterface::class, $secondPage);
+        $this->assertSame($uriGenerator->generate(CategoryApi::CATEGORIES_PATH, [], [
```

Comment:
not a big fan of this kind of test. If we introduce something wrong/useless/regression inside uriGenerator, your test will always be OK. I prefer hardcode url here to have a comparison like that:
`$this->assertSame('http://localhost/api/rest/v1/categories?page=1&limit=2&with_count=false', $secondPage->getPreviousLink())`